Specifications/QIR/Reserved-Operations.md

```diff
@@ -0,0 +1,38 @@
+# Reserved Operation Names
+
+As mentioned [here](Quantum-Runtime.md), we do reserve a set of quantum operation names.
+While targets are not required to implement any or all of these operations,
+if they define an operation with a reserved name that operation's effect is required
+to match the specified action.
+This ensures that quantum code that uses, say, a `CNOT` operation can rely on the
+effect of that operation being consistent across targets that define it.
+
+In this table, operation names should be considered case-insensitive.
+That is, if a target implements `RX` instead of `Rx`, its effect still must match that
+defined in the table.
+
+The matrices in this table below take some time to load.
+Please be patient!
+
+| Operation Name | QIR Runtime Implementation | Description | Matrix | Notes |
+|----------------|-----------------|-------------|-------|-------|
+| Cx, CNOT | [`__quantum__qis__x__ctl`](https://github.com/microsoft/qsharp-runtime/blob/5a379fd1c72ea19cb50d6acc501c6896894e96ac/src/Qir/Runtime/lib/QSharpCore/qsharp-core-qis.ll#L256) | The controlled X operation, aka CNOT operation | ![\begin{bmatrix} 1 & 0 & 0 & 0 \\ 0 & 1 & 0 & 0 \\ 0 & 0 & 0 & 1 \\ 0 & 0 & 1 & 0 \\ \end{bmatrix}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+%5Cbegin%7Bbmatrix%7D+1+%26+0+%26+0+%26+0+%5C%5C+0+%26+1+%26+0+%26+0+%5C%5C+0+%26+0+%26+0+%26+1+%5C%5C+0+%26+0+%26+1+%26+0+%5C%5C+%5Cend%7Bbmatrix%7D) |  |
+| Cz | [`__quantum__qis__z__ctl`](https://github.com/microsoft/qsharp-runtime/blob/5a379fd1c72ea19cb50d6acc501c6896894e96ac/src/Qir/Runtime/lib/QSharpCore/qsharp-core-qis.ll#L282)  | The controlled Z operation | ![\begin{bmatrix} 1 & 0 & 0 & 0 \\ 0 & 1 & 0 & 0 \\ 0 & 0 & 1 & 0 \\ 0 & 0 & 0 & -1 \\ \end{bmatrix}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+%5Cbegin%7Bbmatrix%7D+1+%26+0+%26+0+%26+0+%5C%5C+0+%26+1+%26+0+%26+0+%5C%5C+0+%26+0+%26+1+%26+0+%5C%5C+0+%26+0+%26+0+%26+-1+%5C%5C+%5Cend%7Bbmatrix%7D) |  |
+| H | [`__quantum__qis__h__body`](https://github.com/microsoft/qsharp-runtime/blob/5a379fd1c72ea19cb50d6acc501c6896894e96ac/src/Qir/Runtime/lib/QSharpCore/qsharp-core-qis.ll#L127) | The Hadamard operation | ![\frac{1}{\sqrt{2}}\begin{bmatrix} 1 & 1 \\ 1 & -1 \\ \end{bmatrix}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+%5Cfrac%7B1%7D%7B%5Csqrt%7B2%7D%7D%5Cbegin%7Bbmatrix%7D+1+%26+1+%5C%5C+1+%26+-1+%5C%5C+%5Cend%7Bbmatrix%7D) |  |
+| Measure | [`__quantum__qis__measure__body`](https://github.com/microsoft/qsharp-runtime/blob/5a379fd1c72ea19cb50d6acc501c6896894e96ac/src/Qir/Runtime/lib/QSharpCore/qsharp-core-qis.ll#L140) | The Measure operation |  |
```

Comment:
I would replace this with Mz (measure in the Pauli Z basis). The general, multiqubit Measure operation in Q# is actually pretty unusual, while Mz is standard.

Reply:
Replacing.